integr8ly/grafana-operator
# Customizable config

More flexible configuration of Grafana. This allows users to pass all properties into the generated `grafana.ini` that are supported in the [official documentation](https://grafana.com/docs/installation/configuration/).

The Grafana CR supports a new field, `spec.config`. This is a nested structure that will be serialized into the actual ini file. The names of the sections and keys are exactly the same as described in the Grafana documentation.

Relevant issue: #13 (all the configuration required for this use case should now be supported)

Other changes in this PR:

* Update to operator-sdk version 0.9.0
* Fixes #27 (Datasources provisioning config not created correctly)

Caveats:

* Making changes to existing config should work fine unless whole sections are removed: in that case the operator might receive a stale resource and not apply the update. A workaround is to leave the section header in place when removing properties from `spec.config` until https://github.com/operator-framework/operator-sdk/issues/1700 is resolved

Verification steps:

1. Deploy the operator from this branch
1. Make sure to apply the latest CRDs
1. Create Grafana and make sure it succeeds
1. Compare the config in the CR with the generated configmap `grafana-config`
1. Make a change to the config field in the CR, it should be reflected in the configmap and Grafana should be restarted.
1. Create a data source and make sure the `isDefault` option is set to true (you can use the example one).
1. Log into Grafana and check the datasource: it should be the default one.
